.github/workflows/azure.yaml

```diff
@@ -53,7 +74,41 @@ jobs:
         with:
           creds: ${{ secrets.AZURE_CREDENTIALS }}
 
-      - name: Deploy server code
+      - name: Activate revision if it already exists
+        run: |
+          fetch_app_name=${{ secrets.SETUP_PREFIX }}-fetch
+          nginx_app_name=${{ secrets.SETUP_PREFIX }}-nginx
+          fetch_revision=$fetch_app_name--${{ env.REVISION_ID }}
+          nginx_revision=$nginx_app_name--${{ env.REVISION_ID }}
+
+          az extension add --name containerapp
+
+          if az containerapp revision activate -n $fetch_app_name -g ${{ secrets.RESOURCE_GROUP }} --revision $fetch_revision > /dev/null ; then
+            echo "Fetch revision exists and is activated"
+            fetch_exists=true
+          else
+            echo "Fetch revision is not found or failed to activate"
+            fetch_exists=false
+          fi
+
+          if az containerapp revision activate -n $nginx_app_name -g ${{ secrets.RESOURCE_GROUP }} --revision $nginx_revision > /dev/null ; then
+            echo "Nginx revision exists and is activated"
+            az containerapp ingress traffic set -n $nginx_app_name -g ${{ secrets.RESOURCE_GROUP }} --revision-weight $nginx_revision=100
+            nginx_exists=true
+          else
+            echo "Nginx revision is not found or failed to activate"
+            nginx_exists=false
+          fi
+
+          if [[ $fetch_exists != $nginx_exists ]] ; then
+            echo "Unexpected situation: fetch and nginx apps have different status"
+            (exit 1)
+          fi
```

Comment:
Just let it fail in the next step if this is the case? Imo this kind of "exhaustive errorhandling" belongs in ci-pipelines

Reply:
My comment was more along the line of "do we need to insure integrity?" Or could we just let it run until it fails? As a sidenote: you could check for existence of the variables rather than their value. Saves some initialisation noice at the start of the step.

Reply:
Without this check in error situation where nginx revision doesn't exist but fetch already does, due to setting `REVISION_ALREADY_EXISTS=$fetch_exists`, the revision creation step won't be run. So you will end up with running tests on expected version of fetch and some uknown version of nginx, which is not what you desire. And this won't be discovered.
So for now just check the variables.
Maybe in future, if this Container Apps stands and this becomes too convoluted, I can create more different environments (there were previously only 2 sets per resource group, from last week I think the limit is 5 per subscription) and just always create new revision on every run. [Honestly might have preferred this way from the start if knew what the code looks like, but now I am keen to keep it for experimenting purposes of how stable Container Apps revision managing will be]

Regarding checking for existence of variables: again my bash skills do not let me create something nice out of it because I need to check if both variables are set or unset at the same time. Everything I try is big, scary and fails :smile: This at least seems straightforward.